synapse/rest/client/v2_alpha/account.py

```diff
@@ -499,13 +501,23 @@ class EmailThreepidRequestTokenRestServlet(RestServlet):
     PATTERNS = client_patterns("/account/3pid/email/requestToken$")
 
     def __init__(self, hs):
-        self.hs = hs
         super(EmailThreepidRequestTokenRestServlet, self).__init__()
-        self.identity_handler = hs.get_handlers().identity_handler
+        self.hs = hs
+        self.config = hs.config
         self.store = self.hs.get_datastore()
+        self.identity_handler = hs.get_handlers().identity_handler
 
     @defer.inlineCallbacks
     def on_POST(self, request):
+        if self.config.threepid_behaviour == "off":
```

Comment:
given that we have about 15 copies of this pattern now, how about a method `self.config.assert_threepid_emails_enabled()` which does these checks and raises a generic SynapseError?

Reply:
We could, though we can't log in `*config` so we'd need to do the logging for `local_threepid_handling_disabled_due_to_email_config` outside of that method. So we'd need to check if this method raised an exception, caught it, logged out that "This endpoint was disabled due to a lack of email config block" then bubble the exception.

Reply:
oh, well, we could put it somewhere other than in `*config` then. anyway doesn't matter for now.